Tests/AdyenTests/Adyen Tests/Utilities/BalanceCheckerTests.swift

```diff
@@ -51,28 +58,60 @@ class BalanceValidatorTests: XCTestCase {
+        let result4 = try! sut.check(balance: balance4, isEnoughToPay: amount4)
+        XCTAssertEqual(result4.amountToPay, balance4.availableAmount)
+        XCTAssertEqual(result4.remainingBalanceAmount, .init(value: 0, currencyCode: result4.remainingBalanceAmount.currencyCode))
         XCTAssertFalse(try! sut.check(balance: balance4, isEnoughToPay: amount4).isBalanceEnough)
 
 
-        // transaction limit > balance
-        // amount > balance
+        // balance < amount < transaction limit
         // return false
         let balance5 = Balance(availableAmount: .init(value: 100,
                                                      currencyCode: "EUR"),
                               transactionLimit: .init(value: 1000,
                                                       currencyCode: "EUR"))
         let amount5 = Payment.Amount(value: 120, currencyCode: "EUR")
-        XCTAssertFalse(try! sut.check(balance: balance5, isEnoughToPay: amount5).isBalanceEnough)
+        let result5 = try! sut.check(balance: balance5, isEnoughToPay: amount5)
+        XCTAssertEqual(result5.amountToPay, balance5.availableAmount)
+        XCTAssertEqual(result5.remainingBalanceAmount, .init(value: 0, currencyCode: result5.remainingBalanceAmount.currencyCode))
+        XCTAssertFalse(result5.isBalanceEnough)
 
-        // transaction limit < balance
-        // amount > transaction limit
+        // transaction limit < amount < balance
         // return false
         let balance6 = Balance(availableAmount: .init(value: 1000,
                                                      currencyCode: "EUR"),
                               transactionLimit: .init(value: 100,
                                                       currencyCode: "EUR"))
         let amount6 = Payment.Amount(value: 120, currencyCode: "EUR")
-        XCTAssertFalse(try! sut.check(balance: balance6, isEnoughToPay: amount6).isBalanceEnough)
+        let result6 = try! sut.check(balance: balance6, isEnoughToPay: amount6)
+        XCTAssertEqual(result6.amountToPay, balance6.transactionLimit)
+        XCTAssertFalse(result6.isBalanceEnough)
+        XCTAssertEqual(result6.remainingBalanceAmount, balance6.availableAmount - balance6.transactionLimit)
+
+
+        // transaction limit < balance < amount
+        // return false
+        let balance7 = Balance(availableAmount: .init(value: 1000,
+                                                     currencyCode: "EUR"),
+                              transactionLimit: .init(value: 100,
+                                                      currencyCode: "EUR"))
+        let amount7 = Payment.Amount(value: 1200, currencyCode: "EUR")
+        let result7 = try! sut.check(balance: balance7, isEnoughToPay: amount7)
+        XCTAssertEqual(result7.amountToPay, balance7.transactionLimit)
+        XCTAssertFalse(result7.isBalanceEnough)
+        XCTAssertEqual(result7.remainingBalanceAmount, balance7.availableAmount - balance7.transactionLimit)
```

Comment:
this sort of assumption that is not nice for the tests..

I would rather do:
```swift
XCTAssertEqual(result7.remainingBalanceAmount.value, 900)
XCTAssertEqual(result7.remainingBalanceAmount. currencyCode, "EUR")
```

**+** `fileprivate func - (lhs: Payment.Amount?, rhs: Payment.Amount?) -> Payment.Amount` is not needed